x-pack/filebeat/input/httpjson/input.go

```diff
@@ -297,7 +298,21 @@ func newHTTPClient(ctx context.Context, authCfg *authConfig, requestCfg *request
 		client *http.Client
 		err    error
 	)
-	if authCfg.OAuth2.isEnabled() {
+	if authCfg.AWS.IsEnabled() {
```

Comment:
nit.  Since we now have `else if` any chance I can convince you to switch to `case` (pun intended).